lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_apk_from_aab.rb

```diff
@@ -0,0 +1,132 @@
+        if params[:aab_file_path].nil?
+          all_aab_paths = Actions.lane_context[SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS] || []
+          aab_file_path = if all_aab_paths.count == 1
+                            all_aab_paths.first
+                          else
+                            Actions.lane_context[SharedValues::GRADLE_AAB_OUTPUT_PATH]
+                          end
+        else
+          aab_file_path = params[:aab_file_path]
+        end
+
+        # If no AAB file path was found, raise an error
+        if aab_file_path.nil?
+          UI.user_error!('No AAB file path was specified and none was found in the lane context. Please specify the `aab_file_path` parameter or ensure that the relevant build action has been run prior to this action.')
+          raise
+        end
+
+        apk_output_file_path = params[:apk_output_file_path]
+        keystore_path = params[:keystore_path]
+        keystore_password = params[:keystore_password]
+        keystore_key_alias = params[:keystore_key_alias]
+        signing_key_password = params[:signing_key_password]
+
+        sh(generate_command(aab_file_path, apk_output_file_path, keystore_path, keystore_password, keystore_key_alias, signing_key_password))
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.details
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :aab_file_path,
+            env_name: 'ANDROID_AAB_FILE_PATH',
+            description: 'The path to the AAB file. If not speicified, the action will attempt to read from the lane context using the `SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS` and `SharedValues::GRADLE_AAB_OUTPUT_PATH` keys',
+            type: String,
+            optional: true,
+            default_value: nil,
+            verify_block: proc { |p| UI.user_error!("AAB path `#{p}` is not a valid file path.") unless File.file?(p) }
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :apk_output_file_path,
+            env_name: 'ANDROID_APK_OUTPUT_PATH',
+            description: 'The output path where the APK file will be generated. The directory will be created if it does not yet exist',
+            type: String,
+            optional: false,
+            default_value: nil
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :keystore_path,
+            env_name: 'ANDROID_KEYSTORE_PATH',
+            description: 'The path to the keystore file',
+            type: String,
+            optional: true,
+            default_value: nil,
+            verify_block: proc { |p| UI.user_error!("Keystore file path `#{p}` is not a valid file path.") unless File.file?(p) || p.nil }
```

Comment:
`p.nil` is not a method that exists. (1) The method you want to call is `p.nil?` with a question mark, which is how the method is called to test if a value is nil (it's a convention in Ruby for methods returning boolean results to end with a question mark) and (2) you want to test for nullability **before** trying to test for `File.file?` otherwise `File.file?(nil)` will be tested first and crash due to `nil` parameter.
```suggestion
            verify_block: proc { |p| UI.user_error!("Keystore file path `#{p}` is not a valid file path.") unless p.nil? || File.file?(p) }
```